RFunction.R

```diff
@@ -7,9 +7,110 @@ library('data.table')
 library('sf')
 library('units')
 library('stringr')
+library('tidyr') 
 
 `%!in%` <- Negate(`%in%`)
 
+# Accelerometer Interpolation Function --------------------------------------------
+
+prep_ACC <- function(data, interpolate = FALSE) {
+
+  #' This will interpolate ACC using the nearest two values on the condition that
+  #' both values are within 30mins of the location's timestamp
+  
+  # Unnest into variances ------------------------------
+  
+  # Store track data for later recall
+  tracklevel <- mt_track_data(data)
+  
+  # Split between no ACC and ACC:
+  missrows <- filter(data, is.na(acc_dt)) 
+  accrows <- filter(data, !is.na(acc_dt))  
+
+  # Operate on ACC rows
+  loc_acc_var <- accrows %>% 
+    mutate(
+      var = purrr::map(acc_dt, \(acc_events){
+        if(!is.null(acc_events$acc_burst)){
+          # bind list of matrices (one per ACC event) by row
+          all_bursts <- do.call(rbind, acc_events$acc_burst)
+          # variance in each active ACC axis
+          apply(all_bursts, 2, var)  
+        } else{
+          NULL
+        }
+      })
+    ) %>% 
+    unnest_wider(var, names_sep = "_")
```

Comment:
On a second quick look :p ... quick question: doesn't this chunk of code deals also with row with NULL ACC? I thought it would... If so, I don't think there is a need to split the data (lines 27-28 above) and treat non-acc and acc subsets separately before re-stacking (lines 53 - 58). In other words, wouldn't applying the `mutate` to `data` instead of `accrows` in line 32 lead to an identical dataset as `alldata` in line 53?

Reply:
I'll check this myself during the code review